### PR TITLE
H-45: Make Worker a task and connect them to the Graph

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

We want to allow the workers to connect to the Graph service. To achieve this in the most easy/cheap way I moved the workers next to other containers (such as the Graph itself and Kratos) so the connection can be done by referring to `localhost`.

I kept the `temporal_worker` module to use this again later if we need better scaling.

### :ship: Has this modified a publishable library?

This PR:

- [x] does not modify any publishable blocks or libraries, or modifications do not need publishing

### :scroll: Does this require a change to the docs?

The changes in this PR:

- [x] is internal and does not require a docs change

### :spider_web: Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] does not affect the execution graph

## 📹 Demo

```
❯ temporal workflow execute -t ai --type getDataType --namespace HASH -i '{ "dataTypeId": "https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1" }'
Running execution:
  WorkflowId  <REDACTED>
  RunId       <REDACTED>
  Type        getDataType
  Namespace   HASH
  TaskQueue   ai
  Args        [{"dataTypeId":"https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1"}]

Progress:
  ID          Time                     Type
   1  2023-07-12T13:12:53Z  WorkflowExecutionStarted
   2  2023-07-12T13:12:53Z  WorkflowTaskScheduled
   3  2023-07-12T13:12:53Z  WorkflowTaskStarted
   4  2023-07-12T13:12:53Z  WorkflowTaskCompleted
   5  2023-07-12T13:12:53Z  ActivityTaskScheduled
   6  2023-07-12T13:12:54Z  ActivityTaskStarted
   7  2023-07-12T13:12:54Z  ActivityTaskCompleted
   8  2023-07-12T13:12:54Z  WorkflowTaskScheduled
   9  2023-07-12T13:12:54Z  WorkflowTaskStarted
  10  2023-07-12T13:12:54Z  WorkflowTaskCompleted
  11  2023-07-12T13:12:54Z  WorkflowExecutionCompleted

Result:
  Run Time: 1 seconds
  Status: COMPLETED
  Output: [{"schema":{"$schema":"https://blockprotocol.org/types/modules/graph/0.3/schema/data-type","kind":"dataType","$id":"https://blockprotocol.org/@blockprotocol/types/data-type/text/v/1","title":"Text","description":"An ordered sequence of characters","type":"string"},"metadata":{"recordId":{"baseUrl":"https://blockprotocol.org/@blockprotocol/types/data-type/text/","version":1},"provenance": <REDACTED> }]
```

### CODE REVIEW

File: infra/terraform/hash/hash_application/main.tf [hunk 1]
```diff
@@ -418,6 +430,13 @@ resource "aws_security_group" "app_sg" {
     description = "Allow connections AWS SES"
     cidr_blocks = [var.vpc.cidr_block]
   }
+  egress {
+    from_port   = var.temporal_port
+    to_port     = var.temporal_port
+    protocol    = "tcp"
+    description = "Allow outbound GRPC connections to Temporal"
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (CiaranMn) [inline infra/terraform/hash/hash_application/main.tf:438]: isn't Temporal in the same VPC?
```suggestion
    cidr_blocks = [var.vpc.cidr_block]
```

File: infra/terraform/hash/hash_application/temporal_worker_ai_py.tf [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  tags      = {}
+}
+
+locals {
+  temporal_worker_ai_py_service_container_def = {
+    essential = true
+    name      = local.temporal_worker_ai_py_prefix
+    image     = "${var.temporal_worker_ai_py_image.url}:latest"
+    cpu       = 0 # let ECS divvy up the available CPU
+    dependsOn   = [{ condition = "HEALTHY", containerName = local.temporal_worker_ai_ts_service_container_def.name }]
```
Comment (CiaranMn) [inline infra/terraform/hash/hash_application/temporal_worker_ai_py.tf:25]: Why does the Python worker depend on the TypeScript worker?

**Review by CiaranMn — COMMENTED** [review]
Thanks Tim – a couple of questions

File: infra/terraform/hash/hash_application/temporal_worker_ai_py.tf (hunk 2 above)
Comment (TimDiekmann) [reply infra/terraform/hash/hash_application/temporal_worker_ai_py.tf:25]: I marked it as a dependency as the Python workflows use Typescript workflows but I guess this is solved by temporal itself anyway.
```suggestion
```

**Review by CiaranMn — APPROVED** [review]

**Pull request merged**